mozilla-release/browser/installer/windows/nsis/shared.nsh

```diff
@@ -1349,6 +1349,33 @@ Function FixCliqzAsFirefoxRegistry
   ClearErrors
 FunctionEnd
 
+; Cliqz. Get information about brand from registry and put it into
+; distribution.js file, if it doesn't exist. Here no any checks for
+; data freshness, outdated data must be removed with CliqzHelper.
+Function CliqzSaveBrandingInfo
+  ${If} ${RunningX64}
+    SetRegView 32
+  ${EndIf}
+
+  ; Check for brand information in registry
+  ReadRegStr $0 HKLM "Software\CLIQZ" "${BrandShortName}BrandInfo"
+  ${If} $0 != ""
+    ; Check for distribution.js file, do not replace the existing file.
+    ${IfNot} ${FileExists} "$INSTDIR\defaults\pref\distribution.js"
+      FileOpen $1 "$INSTDIR\defaults\pref\distribution.js" w
+      FileWrite $1 'pref("extensions.cliqz.full_distribution", "$0");'
```

Comment:
I think this way will be better to work with branding information. DLL only save/replace/remove information in registry, all other logic can be controlled from NSIS scripts (no needs to rebuild DLL with all changes)